src/biokbase/narrative/jobs/jobmanager.py

```diff
@@ -477,14 +491,14 @@ def _cancel_job(self, job_id: str) -> None:
         is_refreshing = self._running_jobs[job_id].get("refresh", 0)
         self._running_jobs[job_id]["refresh"] = 0
         self._running_jobs[job_id]["canceling"] = True
-
+        error = None
         try:
             clients.get("execution_engine2").cancel_job({"job_id": job_id})
         except Exception as e:
-            raise transform_job_exception(e, "Unable to cancel job")
-        finally:
-            self._running_jobs[job_id]["refresh"] = is_refreshing
-            del self._running_jobs[job_id]["canceling"]
+            error = transform_job_exception(e, "Unable to cancel job")
```

Comment:
Previously if there was some issue during the processing of a group of job cancellation requests, it would die without finishing the whole list. Changing the functionality so that errors don't cause death and instead collate a list of errors to return with the results.